src/test/java/io/jenkins/plugins/analysis/warnings/AffectedFilesResolverIT.java

```diff
@@ -0,0 +1,419 @@
+                }
+                else {
+                    File file = new File(pathOfFileInWorkspace);
+                    assertThat(file.setReadable(false, false)).isTrue();
+                    assertThat(fileInTmp.setReadable(false, false)).isTrue();
+                }
+            }
+        }
+        String secondPartOfLink = getSourceLink(result);
+        HtmlPage contentPage = webClient.getPage(result.getOwner(),
+                DEFAULT_ENTRY_PATH + firstPartOfLink + secondPartOfLink);
+        checkIfContentExists(contentPage);
+    }
+
+    /**
+     * Verifies that the copied class file is available and can be obtained by HTML scraping.
+     */
+    @Test
+    public void shouldVerifyHtmlOutputBehaviourForAffectedFileWhichShouldFindFile()
+            throws IOException, SAXException {
+        FreeStyleProject project = createJobWithWorkspaceFile(ECLIPSE_TXT, ZIP_FILE);
+        enableWarnings(project);
+        AnalysisResult result = scheduleBuildAndAssertStatus(project, Result.SUCCESS);
+
+        WebClient webClient = createWebClient(false);
+        HtmlPage defaultEntryPage = webClient.getPage(result.getOwner(), DEFAULT_ENTRY_PATH);
+        String firstPartOfLink = getPartOfLink(defaultEntryPage, "fileName", EXTRACTED_FILE);
+        String secondPartOfLink = getSourceLink(result);
+        HtmlPage contentPage = webClient.getPage(result.getOwner(),
+                DEFAULT_ENTRY_PATH + firstPartOfLink + secondPartOfLink);
+        DomElement domElement = contentPage.getElementById("main-panel");
+        DomNodeList<HtmlElement> list = domElement.getElementsByTagName("a");
+        removeElementsByTagName(list);
+
+        String content = convertFileToString(new File(j.jenkins.getWorkspaceFor(project) + "/" + EXTRACTED_FILE));
+
+        assertThat(contentPage.getElementById("main-panel").asText().replaceAll(WHITESPACE, "")).contains(
+                content.replaceAll(WHITESPACE, ""));
+    }
+
+    /**
+     * Verifies that a given file which are copied in a tmp folder can be found by the {@link AffectedFilesResolver}.
+     */
+    @Test
+    public void shouldFindTempFile() {
+        FreeStyleProject project = createJobWithWorkspaceFile(ECLIPSE_TXT, ZIP_FILE);
+        enableWarnings(project);
+        AnalysisResult result = scheduleBuildAndAssertStatus(project, Result.SUCCESS);
+
+        Issues<?> issues = result.getIssues();
+        issues.forEach(issue -> checkIfTmpExists(issue, project, result.getOwner()));
+    }
+
+    /**
+     * Verifies that the AffectedFilesResolver produces an I/O error, when the affected files cannot copied.
+     */
+    @Test
+    public void shouldGetIOErrorBySearchingForAffectedFiles() {
+        FreeStyleProject project = createJobWithWorkspaceFile(ECLIPSE_TXT, ZIP_FILE);
+        enableWarnings(project);
+
+        String pathToExtractedFile = j.jenkins.getWorkspaceFor(project) + "/" + EXTRACTED_FILE;
+        if (System.getProperty("os.name").contains("Windows")) {
```

Comment:
Das Setzen auf ReadOnly wäre in einer Methode schön, dann kann das überall verwendet werden